tests/Export/ProductServiceTest.php

```diff
@@ -421,4 +423,224 @@ public function testProductIsNotFoundIfMainVariantIsNotAvailableForSalesChannel(
+            'id' => $expectedFirstVariantId,
+            'parentId' => $parentId,
+            'productNumber' => 'FINDOLOGIC001.1',
+            'name' => 'FINDOLOGIC VARIANT 1',
+            'price' => [
+                [
+                    'currencyId' => Defaults::CURRENCY,
+                    'gross' => $cheapestPrice,
+                    'net' => $cheapestPrice / 2,
+                    'linked' => false
+                ]
+            ],
+            'options' => [
+                ['id' => $firstOptionId]
+            ],
+        ]));
+
+        $this->createVisibleTestProduct($this->getBasicVariantData([
+            'id' => $expectedSecondVariantId,
+            'parentId' => $parentId,
+            'productNumber' => 'FINDOLOGIC001.2',
+            'name' => 'FINDOLOGIC VARIANT 2',
+            'price' => [['currencyId' => Defaults::CURRENCY, 'gross' => 4, 'net' => 2, 'linked' => false]],
+            'options' => [
+                ['id' => $secondOptionId]
+            ],
+        ]));
+
+        $this->createVisibleTestProduct($this->getBasicVariantData([
+            'id' => $expectedThirdVariantId,
+            'parentId' => $parentId,
+            'productNumber' => 'FINDOLOGIC001.3',
+            'name' => 'FINDOLOGIC VARIANT 3',
+            'price' => [['currencyId' => Defaults::CURRENCY, 'gross' => 4, 'net' => 2, 'linked' => false]],
+            'options' => [
+                ['id' => $thirdOptionId]
+            ],
+        ]));
+
+        // Explicitly assign the main variant ID to the product, and assert that the assigned main variant
+        // is the exported variant.
+        if ($config === 'parent') {
+            $expectedMainVariantId = $expectedThirdVariantId;
+            $this->getContainer()->get('product.repository')->update([
+                [
+                    'id' => $expectedFirstVariantId,
+                    'mainVariantId' => $expectedMainVariantId
+                ],
+                [
+                    'id' => $expectedSecondVariantId,
+                    'mainVariantId' => $expectedMainVariantId
+                ]
+            ], Context::createDefaultContext());
+        }
+
+        // Assert that the cheapest variant is exported as main variant.
+        if ($config === 'cheapest') {
+            $expectedMainVariantId = $expectedFirstVariantId;
+            // If the parent price itself is the cheapest, we export that as the cheapest variant.
+            if ($parentPrice <= $cheapestPrice) {
+                $expectedMainVariantId = $parentId;
+            }
+        }
+
+        $mockedConfig = $this->getFindologicConfig(['mainVariant' => $config]);
+        $mockedConfig->initializeBySalesChannel($this->salesChannelContext);
+
+        $this->defaultProductService->setConfig($mockedConfig);
+        $result = $this->defaultProductService->searchVisibleProducts(20, 0);
+        $elements = $result->getElements();
+
+        $this->assertNotEmpty($elements);
```

Comment:
The moment you assert that the count is at least `1`, you're already ensuring that this is not empty, so I assume this check isn't really necessary. This check can also probably be removed from `testProductWithoutVariantsBasedOnExportConfig`.